cpp/src/parquet/statistics.cc

```diff
@@ -472,34 +474,40 @@ class TypedStatisticsImpl : public TypedStatistics<DType> {
     comparator_ = std::static_pointer_cast<TypedComparator<DType>>(comp);
     TypedStatisticsImpl::Reset();
     has_null_count_ = true;
-    has_distinct_count_ = true;
+    has_distinct_count_ = false;
```

Comment:
nit: add a comment here to explain why has_null_count_ and has_distinct_count_ have different defaults as we have discussed here